drv/lpc55-sprot-server/src/handler.rs

```diff
@@ -568,8 +568,46 @@ impl<'a> Handler {
                 )?;
                 Ok((RspBody::Ok, None))
             }
+            ReqBody::State(StateReq::DevOrRelease) => {
+                let out = self.state_dev_or_release();
+
+                Ok((RspBody::State(out.map(StateRsp::DevOrRelease)), None))
+            }
         }
     }
+
+    fn state_dev_or_release(
+        &mut self,
+    ) -> Result<StateDevOrRelease, StateError> {
+        let mut buf = [0u8; 512];
+
+        // If the SHA-256 Digest is zeros, the CMPA is unlocked
+        self.update
+            .read_rot_page(RotPage::Cmpa, &mut buf)
+            .map_err(StateError::ReadCmpa)?;
+        let unlocked = buf[480..].iter().all(|b| *b == 0);
+
+        // Otherwise, check which root keys are enabled
+        let dev = if unlocked {
+            true
+        } else {
+            self.update
+                .read_rot_page(RotPage::CfpaActive, &mut buf)
+                .map_err(StateError::ReadCfpa)?;
+            // Look at the ROTKH_REVOKE byte
+            let revoke = buf[24];
+            // Check if any of the development keys (slots 2 and 3) are
+            // in the Valid state (0b01)
+            [2, 3]
```

Comment:
huh, cute

Reply:
Do we have symbolic names for those slots anywhere?

Reply:
Not that I can find, but I think the comment above is sufficient to explain their meaning.